osu.Game/Online/Chat/ChannelManager.cs

```diff
@@ -445,19 +445,39 @@ public void JoinLastClosedChannel()
                 return;
             }
 
-            Channel lastClosedChannel = closedChannels.Last();
+            // This nested loop could be eliminated if a check was added so that
+            // when the code opens a channel it removes from the closedChannel list.
+            // However, this would require adding an O(|closeChannels|) work operation
+            // every time the user joins a channel, which would make joining a channel
+            // slower. I wanted to centralize all major slowdowns so they
+            // can only occur if the user actually decides to use this feature.
+            for (int i = closedChannels.Count - 1; i >= 0; i--)
+            {
+                string lastClosedChannelName = closedChannels.Last();
+                closedChannels.RemoveAt(closedChannels.Count - 1);
+                bool alreadyJoined = false;
 
-            closedChannels.Remove(lastClosedChannel);
+                // If the user already joined the channel, do not
+                // try to join it
+                for (int j = 0; j < joinedChannels.Count; j++)
```

Comment:
I think this can be condensed to using a FirstOrDefault with a lookup argument - search "available" in this file for an examplee